### PR TITLE
fix: log router_key in web/webpush web/simplepush

### PR DESCRIPTION
**Pull request by jrconlin**
closes: #745

### CODE REVIEW

File: autopush/router/webpush.py [hunk 1]
```diff
@@ -132,6 +132,11 @@ def _save_notification(self, uaid, notification, month_table):
                                   headers={"TTL": str(notification.ttl),
                                            "Location": location},
                                   logged_status=204)
+        self.log.info("Saving message data",
```
Comment (bbangert) [inline autopush/router/webpush.py]: We already log whether it was stored or online delivery in web/webpush.py, shouldn't the size and such be added there?

File: autopush/web/webpush.py [hunk 2]
```diff
@@ -382,18 +385,25 @@ def _router_completed(self, response, uaid_data, warning=""):
...
+                response,
+                uaid_data,
+                warning,
+                notification))
             return d
         else:
             # No changes are requested by the bridge system, proceed as normal
             if response.status_code == 200 or response.logged_status == 200:
                 self.log.info(format="Successful delivery",
                               client_info=self._client_info)
```
Comment (bbangert) [inline autopush/web/webpush.py]: Need all the data here as well.

Comment (jrconlin) [inline autopush/web/webpush.py]: Cool, wasn't sure if you did.

File: autopush/web/webpush.py [hunk 3]
```diff
@@ -382,18 +385,25 @@ def _router_completed(self, response, uaid_data, warning=""):
...
             elif response.status_code == 202 or response.logged_status == 202:
-                self.log.info(format="Router miss, message stored.",
-                              client_info=self._client_info)
+                self.log.info(
+                    format="Router miss, message stored.",
+                    message_size=len(notification.data) if notification else 0,
```
Comment (pjenvey) [inline autopush/web/webpush.py]: This was _client_info['message_size'] in endpoint. let's just dump the notification metadata in there up front so we don't have to the notification around

File: autopush/web/webpush.py [hunk 4]
```diff
@@ -382,18 +385,25 @@ def _router_completed(self, response, uaid_data, warning=""):
...
+                    ttl=notification.ttl if notification else 0,
+                    uaid=uaid_data['uaid'],
```
Comment (pjenvey) [inline autopush/web/webpush.py]: uaid should already be in client_info

**Comment by bbangert** [comment]
Can we add the timing to the two log statements? It's not really feasible to get the data out of DataDog, so it'd be handy to include it with the message here (which would also let us easily differentiate time to store vs. online delivery).

**Comment by jrconlin** [comment]
Added custom log observer (using a callback to do the actual testing).

Added route time to logging https://github.com/mozilla-services/autopush/pull/747/files#diff-ccbbd52ff880cf4ddb71b18d126d189cR369

File: autopush/web/webpush.py [hunk 5]
```diff
@@ -382,18 +388,20 @@ def _router_completed(self, response, uaid_data, warning=""):
...
             if response.status_code == 200 or response.logged_status == 200:
                 self.log.info(format="Successful delivery",
                               client_info=self._client_info)
             elif response.status_code == 202 or response.logged_status == 202:
-                self.log.info(format="Router miss, message stored.",
-                              client_info=self._client_info)
+                self.log.info(
+                    format="Router miss, message stored.",
+                    client_info=self._client_info)
             time_diff = time.time() - self.start_time
```
Comment (bbangert) [inline autopush/web/webpush.py]: Why not just move this up and log this time?

**Pull request merged**